src/main/java/com/avispl/symphony/dal/device/core510i/QRCCommunicator.java

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright (c) 2021 AVI-SPL, Inc. All Rights Reserved.
+ */
+package com.avispl.symphony.dal.device.core510i;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.util.Arrays;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import com.avispl.symphony.api.dal.dto.control.ConnectionState;
+import com.avispl.symphony.api.dal.error.CommandFailureException;
+import com.avispl.symphony.dal.BaseDevice;
+import com.avispl.symphony.dal.communicator.Communicator;
+import com.avispl.symphony.dal.communicator.ConnectionStatus;
+
+/**
+ * An implementation of QRCCommunicator to provide communication and interaction with QSys-Core
+ *
+ * @author Harry
+ * @since 1.0
+ */
+public class QRCCommunicator extends BaseDevice implements Communicator {
+	private static final String ERROR_MESSAGE_CHANGE_PROPERTIES_AFTER_INIT = "Cannot change properties after init() was called";
+	private final ReadWriteLock lock = new ReentrantReadWriteLock();
+	private final ConnectionStatus status = new ConnectionStatus();
+	protected String login;
+	protected String password;
+	protected int numOfResponses = 2;
+	private int socketTimeout = 4000;
+	private Socket socket;
+	private int port;
+
+	/**
+	 * This method returns the device UPD port
+	 *
+	 * @return int This returns the current UDP port.
+	 */
+	public int getPort() {
+		return this.port;
+	}
+
+	/**
+	 * This method is used set the device UDP port
+	 *
+	 * @param port This is the UDP port to set
+	 */
+	public void setPort(int port) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException("Cannot change properties after init() was called");
```

Comment:
You already have the constant ERROR_MESSAGE_CHANGE_PROPERTIES_AFTER_INIT on the top, please use it!

Reply:
thank you, I forgot it :v